### ISSUE TITLE
[bug] pandas ValueError: numpy.dtype has the wrong size, try recompiling

### ISSUE THREAD

**Issue by poppingtonic** [probcomp/bayeslite#376]
I compiled bayeslite under a `Python 2.7.6` environment and the tests `./check.sh` run ok for my current `master` branch. All except `tests/test_read_pandas.py`:

```bash
tests/test_read_pandas.py:18: in <module>
    import pandas
../../.envs/bayeslite/local/lib/python2.7/site-packages/pandas/__init__.py:7: in <module>
    from pandas import hashtable, tslib, lib
pandas/src/numpy.pxd:157: in init pandas.hashtable (pandas/hashtable.c:38262)
    ???
E   ValueError: numpy.dtype has the wrong size, try recompiling

```

I also see this when I run `bayesdb-demo` in my virtualenv. Installing the latest pandas and numpy doesn't fix the problem.

**Comment by riastradh-probcomp** [comment]
It sounds like the versions of numpy and pandas that you have installed don't match, which is a problem outside bayeslite.

How did you install numpy and pandas?  What operating system are you using?

Generally I would recommend either

(a) ensuring that all the dependencies are installed yourself, e.g. via apt-get install; or
(b) installing bayeslite and its dependencies with pip, inside a virtual env.

**Comment by gregory-marton** [comment]
virtualenv is not enough in this case, because pip might still use an incompatible cached binary. You need to tell it to recompile. That's why I suggested (ooh, I suggested this in flowdoc, so it didn't show up here) doing `pip --no-cache-dir install pandas` once you have the numpy version you want.

**Comment by poppingtonic** [comment]
@riastradh-probcomp I'm using Ubuntu 14.04.3 LTS. I also created a virtualenv based on `python 2.7.6` and installed bayeslite + deps into it. 

@gregory-marton I'll try that and see what happens. I also assume that I should make sure to install `pandas==0.13.1`.

**Comment by gregory-marton** [comment]
@poppingtonic how did that go?

What's specified in setup.py is correct, and for bayeslite alone does not specify a desired pandas version. 
https://github.com/probcomp/bayeslite/blob/master/setup.py#L188

For the recommended bdbcontrib package that accompanies it, we do specify a pandas version, and the one you quote seems too old:
https://github.com/probcomp/bdbcontrib/blob/master/setup.py#L131

Going up a level, what's your end goal?

**Comment by gregory-marton** [comment]
ping?

**Comment by poppingtonic** [comment]
Oh, I got very busy and completely forgot about this. So sorry :(. My end goal is to get a feel for how bayeslite's query language works. I'm fascinated by the crosscat and bayeslite and would like to learn how they work, Hopefully I can contribute once that happens. So what I did was install `pandas==0.17.1` from source, and build `bdbcontrib` as described in the repo, using `python setup.py build`. After that, my tests were all green.
